### PR TITLE
feat(limit-* plugin):  fallback to remote_addr when key is missing

### PR DESCRIPTION
**Pull request by Xunzhuo**
### What this PR does / why we need it:

 fallback to `remote_addr` when key is missing or invalid in limit-* plugin

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/plugins/limit-conn/init.lua [hunk 1]
```diff
@@ -64,10 +64,9 @@ function _M.increase(conf, ctx)
     end
 
     if key == nil then
-        core.log.info("bypass the limit conn as the key is empty")
-        -- Bypass the limit conn when the key is empty.
-        -- This behavior is the same as Nginx
-        return
+        core.log.info("The key of limit conn is empty or invalid, set $remote_addr as the key")
+        -- When the key is empty or invalid, set $remote_addr as the default key
+        key = "remote_addr"
```
Comment (spacewander) [inline apisix/plugins/limit-conn/init.lua:69]: The key should be the value of remote_addr not remote_addr itself

Comment (Xunzhuo) [reply apisix/plugins/limit-conn/init.lua]: Done. Thanks

File: t/plugin/limit-conn2.t [hunk 2]
```diff
@@ -410,32 +440,102 @@ GET /t
...
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/limit_conn"
+                }]],
+                [[{
```
Comment (spacewander) [inline t/plugin/limit-conn2.t:486]: There is no need to check the response, right?

File: t/plugin/limit-conn2.t [hunk 3]
```diff
@@ -321,8 +351,8 @@ request latency is nil
                  [[{
                         "plugins": {
                             "limit-conn": {
-                                "conn": 2,
-                                "burst": 1,
+                                "conn": 1,
```
Comment (spacewander) [inline t/plugin/limit-conn2.t:354]: Why change this configuration?

File: docs/en/latest/plugins/limit-conn.md [hunk 4]
```diff
@@ -42,14 +42,14 @@ Limiting request concurrency plugin.
...
 | key_type      | string  | optional    |   "var"   | ["var", "var_combination"] | the type of key. |
-| key           | string  | required    |         |  | the user specified key to limit the rate. If the `key_type` is "var", the key will be treated as a name of variable, like "remote_addr" or "consumer_name". If the `key_type` is "var_combination", the key will be a combination of variables, like "$remote_addr $consumer_name". |
+| key           | string  | required    |         |  | the user specified key to limit the rate. If the `key_type` is "var", the key will be treated as a name of variable, like "remote_addr" or "consumer_name". If the `key_type` is "var_combination", the key will be a combination of variables, like "$remote_addr $consumer_name". If the key is empty or invalid, `remote_addr` will be set as the default key.|
```
Comment (spacewander) [inline docs/en/latest/plugins/limit-conn.md:45]:
```suggestion
| key           | string  | required    |         |  | the user specified key to limit the rate. If the `key_type` is "var", the key will be treated as a name of variable, like "remote_addr" or "consumer_name". If the `key_type` is "var_combination", the key will be a combination of variables, like "$remote_addr $consumer_name". If the value of the key is empty, `remote_addr` will be set as the default key.|
```

File: apisix/plugins/limit-req.lua [hunk 5]
```diff
@@ -102,10 +102,9 @@ function _M.access(conf, ctx)
     end
 
     if key == nil then
-        core.log.info("bypass the limit req as the key is empty")
-        -- Bypass the limit req when the key is empty.
-        -- This behavior is the same as Nginx
-        return
+        core.log.info("The key of limit req is empty or invalid, set $remote_addr as the key")
```
Comment (spacewander) [inline apisix/plugins/limit-req.lua:105]:
```suggestion
        core.log.info("The value of the configured key is empty, use client IP instead")
```

The value is empty, not the key itself.

Comment (Xunzhuo) [reply apisix/plugins/limit-req.lua:105]: Done, thanks.

File: docs/en/latest/plugins/limit-conn.md (hunk 4 above)
Comment (Xunzhuo) [reply docs/en/latest/plugins/limit-conn.md:45]: Done, thanks.

File: t/plugin/limit-conn2.t (hunk 2 above)
Comment (Xunzhuo) [reply t/plugin/limit-conn2.t:486]: Sorry, I do not understand this point? Could you give me more infos?

File: t/plugin/limit-conn2.t (hunk 3 above)
Comment (Xunzhuo) [reply t/plugin/limit-conn2.t:354]: It is easier to test when value of key is empty.

File: t/plugin/limit-conn2.t (hunk 2 above)
Comment (Xunzhuo) [reply t/plugin/limit-conn2.t:486]: Do you mean that there is no need to check this?
```
--- response_body
passed
```

File: t/plugin/limit-conn2.t (hunk 3 above)
Comment (spacewander) [reply t/plugin/limit-conn2.t:354]: Why changing the configuration doesn't provide a different result?

Comment (spacewander) [reply t/plugin/limit-conn2.t:354]: I see. It is already covered in the new test.

**Review by spacewander — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**